### ISSUE TITLE
template-implements on DOMNodeList stub file does not appear to take effect

### ISSUE THREAD

**Issue by SignpostMarv** [vimeo/psalm#1284]
* typed generator example: https://getpsalm.org/r/35af8b3de1
* `array<class-string<DOMNode>, int>` example: https://getpsalm.org/r/6a1103b920
* using 6d5c995

psalm currently requires `$item` to be type-hinted, despite the [stub entry in `CoreGenericClasses.php` for `DOMNodeList`](https://github.com/vimeo/psalm/blob/6d5c99545874de2cf167b88a993f5fda6783cef2/src/Psalm/Internal/Stubs/CoreGenericClasses.php#L725) appearing to be templated as expected.

Since we "know" that `DOMNodeList` implements `Traversable<int, TNode as DOMNode>`, psalm would preferably not identify `foreach($list as $item)` as having `$item` typed as `mixed`, but instead whatever `TNode` was.

```php
/**
* @template T as DOMNode
*
* @param class-string<T> $type
*
* @return Generator<int, T, mixed, int>
*/
function Bar(DOMNodeList $list, string $type) : Generator {
  $matching = 0;

  foreach ($list as $item) {
    if (is_a($item, $type)) {
      $matching += 1;
      
      yield $item;
    }
  }
  
  return $matching;
}
```

**Comment by muglug** [comment]
Fixed in c83afc1af21f0d434d066e80779d8ef5f0f049ca

**Comment by SignpostMarv** [comment]
:D
